### PR TITLE
Fix typestack validateBody hook

### LINKED ISSUE #785: Error in typestack ValidateBody
```typescript
import { ValidateBody } from '@foal/typestack'

abstract class EntityController {
  protected constructor(
    protected entityClass: Class<I>
  ) {
  }  

  @ValidateBody((controller: EntityController<any>) => controller.entityClass)
  async create(ctx: Context) {}
}

class SomeController extends EntityController {
  constructor() {
    super(SomeEntity)
  }
}

class AnotherController extends EntityController {
  constructor() {
    super(AnotherEntity)
  }
}
```

When you call create for one repository validator is fixed to SomeEntity for example.

Error is in  '@foal/typestack' ValidateBody `cls = isArrowFunction(cls) ? cls(this) : cls;`.
Function parameter is overriden on the first call.
Spotted by @vedyakov.

### PR DESCRIPTION
**Pull request by AleksandrSl**
Fixes #785

# Checklist

- [x] Add/update/check docs (code comments and docs/ folder).
- [x] Add/update/check tests.
- [ ] Update/check the cli generators.

I'm not sure what the target branch should be

### CODE REVIEW

File: packages/typestack/src/validate-body.hook.ts [hunk 1]
```diff
@@ -38,5 +39,5 @@ export function ValidateBody(
 }
 
 function isArrowFunction(value: any): value is ((controller: any) => Class) {
-  return !value.hasOwnProperty('prototype');
+  return !Object.prototype.hasOwnProperty.call(value, 'prototype');
```
Comment (LoicPoullain) [inline packages/typestack/src/validate-body.hook.ts:42]: Why this change?

Comment (AleksandrSl) [reply packages/typestack/src/validate-body.hook.ts:42]: Just to be sure that we won't break, if someone gives us object with builtin properties messed up. Maybe it's overkill
